src/app/[bookId]/page.tsx

```diff
@@ -0,0 +1,46 @@
+'use client'
+
+import { useState, useEffect } from 'react'
+
+import BooksAPI from '@/api/books-api'
+import BookInfo from '@/app/[bookId]/components/book-info'
+import BookLikeButton from '@/app/[bookId]/components/book-like-button'
+import BookOpinionTabs from '@/app/[bookId]/components/book-opinion-tabs'
+import type { Book } from '@/types/book'
+
+interface BookProps {
+  params: { bookId: string }
+}
+
+const Book = ({ params }: BookProps) => {
+  const [book, setBook] = useState<Book | null>(null)
+
+  const fetchBook = async () => {
+    const fetchedBook = await BooksAPI.getBookById(params.bookId)
+    setBook(fetchedBook)
+  }
+
+  useEffect(() => {
+    fetchBook()
+  }, [params.bookId])
+
+  if (!book) {
+    return <div>Книгу не знайдено</div>
```

Comment:
```suggestion
    return (
      <div className='flex flex-col items-center justify-center h-[600px] text-center  '>
        <Loader2 className=' self-center w-12 h-12 animate-spin text-gray-300' />
        <span className='text-2xl text-gray-300 font-bold mb-2'>
          Завантаження...
        </span>
      </div>
    )
```
here it needs some design

Reply:
Added